src/embind/emval.js

```diff
@@ -126,6 +126,15 @@ var LibraryEmVal = {
     return Emval.toHandle([]);
   },
 
+  _emval_new_array_from_memory_view__deps: ['$Emval'],
+  _emval_new_array_from_memory_view: function(view) {
+    view = Emval.toValue(view);
+    var a = [];
```

Comment:
Preallocating array via `new Array(view.length)` might make it a bit faster yet. That said, I wonder what the difference is like between `Array.from` and `for-loop`? I was under impression that in modern V8 they should be pretty close.

Reply:
> I wonder what the difference is like between `Array.from` and `for-loop`?

I don't know why but I did some tests and found a for...loop is most fast, then .slice(), Array.from()..
Emm, preallocating might be a bit faster more.

Reply:
> I don't know why but I did some tests and found a for...loop is most fast, then .slice(), Array.from()..

Oh know, I know why but I meant literally what's the difference in terms of % of speed.

Reply:
I remember it was about 20x vs. 5x (for..loop vs Array.from) in the test.  I'll confirm it later.

Reply:
Vow!  Preallocating array does faster about 24%! (38 vs 50)

```
+val::array: 1024.721900
+val::array opt numeric types: 38.470097
```

Using Array.from().

```
+val::array: 1004.742680
+val::array opt numeric types: 201.159535
```
Curious why Array.from() is slow so much?

Reply:
> Preallocating array does faster about 24%!

Nice!

> Curious why Array.from() is slower so much?

Array.from is meant to be generic over any iterables, so it uses the iterable protocol (by getting an iterator `Symbol.iterator` on whatever you pass it and repeatedly calling `.next()` on it) which can't be optimized as well by the compiler as plain loops + indexed access. I thought V8 had specializations for common sources, but apparently not for Typed Arrays.

One more thing you can try if you want is using `Array.prototype.slice.call(arr)` - it's probably still going to be a bit slower than plain loop, but should be still much faster than `Array.from` here.

Reply:
> using `Array.prototype.slice.call(arr)` - it's probably still going to be a bit slower than plain loop, but should be still much faster than `Array.from` here.

I did. The result is extract as what you said.  A plain for loop is fastest, then slice() and [].concat().